x-pack/platform/plugins/shared/alerting/server/lib/rule_gaps/task/gap_auto_fill_scheduler_task.ts

```diff
@@ -0,0 +1,404 @@
+                });
+                return { state: {} };
+              }
+
+              // Step 3: Fetch rule IDs with gaps
+              let remainingBackfills = capacityCheckInitial.remainingCapacity;
+              // newest gap first
+              const sortOrder = 'desc';
+              const { ruleIds } = await rulesClient.getRuleIdsWithGaps({
+                start: startDate.toISOString(),
+                end: now.toISOString(),
+                sortOrder,
+                hasUnfilledIntervals: true,
+              });
+
+              if (!ruleIds.length) {
+                await logEvent({
+                  status: GAP_AUTO_FILL_STATUS.SKIPPED,
+                  results: [],
+                  message: 'Skipped execution: no rules with gaps',
+                });
+                return { state: {} };
+              }
+
+              const aggregatedByRule = new Map<
+                string,
+                {
+                  ruleId: string;
+                  processedGaps: number;
+                  status: GapFillSchedulePerRuleStatus;
+                  error?: string;
+                }
+              >();
+
+              // Step 4: Process rules in batches
+              for (
+                let startIdx = 0;
+                startIdx < ruleIds.length;
+                startIdx += DEFAULT_RULES_BATCH_SIZE
+              ) {
+                // Stop early if we've reached capacity
+                if (remainingBackfills <= 0) {
+                  const consolidated = resultsFromMap(aggregatedByRule);
+                  await logEvent({
+                    status: GAP_AUTO_FILL_STATUS.SUCCESS,
+                    results: consolidated,
+                    message: `Stopped early: no backfill capacity remaining | ${formatConsolidatedSummary(
+                      consolidated
+                    )}`,
+                  });
+                  return { state: {} };
+                }
+                if (
+                  await handleCancellation({
+                    abortController,
+                    aggregatedByRule,
+                    logEvent,
+                  })
+                ) {
+                  return { state: {} };
+                }
+
+                const currentRuleIds = ruleIds.slice(startIdx, startIdx + DEFAULT_RULES_BATCH_SIZE);
+
+                // we need to find the rules that are enabled and
+                // and check for rule types and consumers that are supported
+                const { data: rules } = await rulesClient.find({
+                  options: {
+                    page: 1,
+                    perPage: currentRuleIds.length,
+                    filter: `alert.attributes.enabled:true AND (${currentRuleIds
+                      .map((id) => `alert.id: ("alert:${id}")`)
+                      .join(' OR ')})`,
+                    ruleTypeIds: config.ruleTypes.map((rt) => rt.type),
```

Comment:
Seems like this is something we could add to the `getRuleIdsWithGaps` function to filter by rule type and consumer in that function and only return the rule IDs that match that have gaps?